catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/watcher/FileWatcher.java

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.content.monitor.watcher;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+import org.codice.ddf.platform.util.StandardThreadFactoryBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Monitors the size of files. When a file's length is considered "stable" (non changing between
+ * poll times), the file's corresponding callback will be called.
+ */
+public class FileWatcher {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(FileWatcher.class);
+
+  private static final long DEFAULT_POLL_TIME_SEC = 5;
+
+  private static final String CDM_FILE_CHECK_PERIOD_PROPERTY = "org.codice.ddf.cdm.fileCheckPeriod";
+
+  private Map<File, Consumer<File>> fileMap;
+
+  private Map<File, Long> fileSizeMap;
+
+  private ScheduledExecutorService executorService;
+
+  private long executorPollTime;
+
+  /**
+   * Creates a FileWather with default poll time as defined by the {@code
+   * org.codice.ddf.cdm.fileCheckPeriod} system property, or 5 seconds if it is not defined.
+   */
+  public FileWatcher() {
+    this.executorPollTime = getPollTime();
+    fileMap = new HashMap<>();
+    fileSizeMap = new HashMap<>();
+    init();
+  }
+
+  /**
+   * Creates a FileWatcher with the given poll time.
+   *
+   * @param pollTime pollTime in seconds
```

Comment:
✏️ Nit: _polling interval_ instead of _pollTime_. We could also consider renaming the parameter and field to `pollingInterval`.